### DISCUSSION TITLE
support shared caches (e.g. redis) in sqlalchemy 1.4 query cache

### DISCUSSION THREAD

**Discussion by the2nd** [sqlalchemy/sqlalchemy#5854] [Usage Questions]
Will the new query cache in v1.4 support shared caches like redis? My project does use multiprocessing because of the GIL. Caching the query result works fine and it would be nice to share the cached sql statements between processes too.

**Comment by zzzeek** [comment]
hey there -

The query cache does not cache database results, it is used only as an internal optimization to cache the **compiled SQL structure** that is first passed to the database.   this feature is reducing the work of Python overhead in scanning through the Python object structure that represents a statement and formulating a string SQL statement, as well as calculating other object structures that are used to process result set rows.  

This cache would not be appropriate outside of the process using a networked service such as Redis.  To be able to receive these object structures from redis would require that they be fully serializable and that the entire object structure be built up from the serialized string in any case, which is at best marginally more peformant than the orgiinal string compilation task if at all, and would continue to be vastly slower than having the structure available as its original Python structure in memory.  this cache is more of a Python object generation optimization and not any kind of optimization for acquiring raw data.
